src/picongpu/include/particles/flylite/NonLTE.hpp

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2017 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "particles/flylite/NonLTE.def"
+#include "simulation_defines.hpp"
+#include "particles/flylite/IFlyLite.hpp"
+
+/* PMacc */
+#include "dimensions/DataSpace.hpp"
+
+#include <memory>
+
+
+namespace picongpu
+{
+namespace particles
+{
+namespace flylite
+{
+    template<
+        //! @todo for multi ion species IPD: typename T_OtherIonsList,
+        typename T_ElectronsList,
+        typename T_PhotonsList
+    >
+    class NonLTE : public IFlyLite
+    {
+    public:
+        //! @todo for multi ion species IPD: using OtherIonsList = T_OtherIonsList;
+
+        using ElectronsList = T_ElectronsList;
+        using PhotonsList = T_PhotonsList;
+
+        virtual
+        void
+        init(
+            PMacc::DataSpace< simDim > const & gridSizeLocal,
+            std::string const ionSpeciesName
```

Comment:
Sry I haven't seen it in the first iteration but this should be a reference `std::string const & ionSpeciesName`